doc/nimc.rst

```diff
@@ -326,20 +326,66 @@ They are:
 DLL generation
 ==============
 
-Nim supports the generation of DLLs. However, there must be only one
-instance of the GC per process/address space. This instance is contained in
-``nimrtl.dll``. This means that every generated Nim DLL depends
-on ``nimrtl.dll``. To generate the "nimrtl.dll" file, use the command::
+Nim supports the generation of dynamic libraries. However, there must be only
+one instance of the GC per process/address space, meaning that if multiple Nim
+DLLs are used by the same binary, the DLLs must link to the ``nimrtl`` dynamic 
+library.
+
+**Note for UNIX**: If any Nim DLL (including ``libnimrtl.so`` and
+``libnimchr.so``) is to be directly included with your application in the same
+folder as the app binary, the app's ``rpath`` variable should be set to
+``$ORIGIN``. The reason for this is because the dynamic libary linker for
+UNIX-like systems *does not* search for dynamic libaries in the same folder
+(unlike Windows), unless the ``rpath`` variable is set otherwise in the binary
+to search for in the app directory.
+
+For an app binary built in Nim, an easy way to set the ``rpath`` variable
+correctly is by including the following line in your source code:
 
-  nim c -d:release lib/nimrtl.nim
+.. code-block:: Nim
+  when defined(unix): {.passL: "-Wl,-rpath='$ORIGIN'"}
+
+``nimrtl``
+----------
+
+The ``nimrtl`` libarary contains both Nim's memory manager as well as the full
+standard libary. To generate the ``nimrtl`` library, first, create a file named
+``build_nimrtl.nim``, and fill it with the following:
+
+.. code-block:: Nim
+  include nimrtl
+
+Then, use the command::
+
+  nim c -d:release -d:createNimRtl --app:lib --out:nimrtl build_nimrtl.nim
 
-To link against ``nimrtl.dll`` use the command::
+Rename ``nimrtl`` to ``nimrtl.dll`` (for Windows) or ``libnimrtl.so`` (for UNIX)
+accordingly to your platform.
+
+To link against ``nimrtl`` use the command::
 
   nim c -d:useNimRtl myprog.nim
 
 **Note**: Currently the creation of ``nimrtl.dll`` with thread support has
 never been tested and is unlikely to work!
 
+``nimhcr``
+----------
+Nim's hot-code-reloading support requires the app binary to link to both
+``nimrtl`` as well as another library, ``nimchr``. Similarly to how ``nimrtl``
+was built, you can generate this library by create a file named
+``build_nimhcr.nim``, filling it with the following:
+
+.. code-block:: Nim
+  include nimhcr
+
+Compiling with the command::
+
+  nim c -d:release -d:createNimHcr --app:lib --out:nimrtl build_nimrtl.nim
```

Comment:
`--out:nimrtl build_nimrtl.nim` should be `--out:nimhcr build_nimhcr.nim` instead.